src/engine/controls/bpmcontrol.cpp

```diff
@@ -725,6 +726,160 @@ double BpmControl::getNearestPositionInPhase(
     return dNewPlaypos;
 }
 
+double BpmControl::getBeatMatchPosition(
+        double dThisPosition, bool respectLoops, bool playing) {
+    // Without a beatgrid, we don't know the phase offset.
+    if (!m_pBeats) {
+        return dThisPosition;
+    }
+    // Master buffer is always in sync!
+    if (getSyncMode() == SYNC_MASTER) {
```

Comment:
There's one case where this isn't true: if this deck has master sync on and the other deck does not, and is playing.  When we press play, we expect the new deck to sync phase to the other non-sync deck. Luckily, the fix for this should just be to remove this clause.

Reply:
(It looks like this code is copied from existing code, check out my mastersync-onemaster branch to see how it changed)

Reply:
I am afraid that just removing this clause will break other edge cases. 
If the track is already playing and is SYNC_Master, than this condition is probably still desired. 
I will check.